Multisig/Logic/Models/WCSession.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  WCSession.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 22.01.21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import CoreData
+import WalletConnectSwift
+
+extension WCSession {
+    enum WCSessionStatus: Int {
+        case connecting = 0, connected
+    }
+
+    var status: WCSessionStatus {
+        get {
+            WCSessionStatus(rawValue: Int(statusRaw))!
+        } set {
+            statusRaw = Int16(newValue.rawValue)
+        }
+    }
+
+    static func getAll() throws -> [WCSession] {
+        do {
+            let context = App.shared.coreDataStack.viewContext
+            let fr = WCSession.fetchRequest().all()
+            let sessions = try context.fetch(fr)
+            return sessions
+        } catch {
+            throw GSError.DatabaseError(reason: error.localizedDescription)
+        }
+    }
+
+    static func get(topic: String) -> WCSession? {
+        let context = App.shared.coreDataStack.viewContext
+        let fr = WCSession.fetchRequest().by(topic: topic)
+        return try? context.fetch(fr).first
+    }
+
+    static func create(wcurl: WCURL) {
+        dispatchPrecondition(condition: .onQueue(.main))
+        let context = App.shared.coreDataStack.viewContext
+        let wcSession = WCSession(context: context)
+        wcSession.status = .connecting
+        wcSession.created = Date()
+        wcSession.topic = wcurl.topic
```

Comment:
we should create or update sessions with the same topic

Reply:
not crucial as we cleanup stale session, but ok, I will change it